corehq/apps/locations/bulk_management.py

```diff
@@ -421,7 +431,11 @@ def bulk_commit(self, type_stubs, location_stubs):
             loc.lookup_old_collection_data(self.old_collection)
 
         type_objects = save_types(type_stubs, self.excel_importer)
-        save_locations(location_stubs, type_objects, self.domain, self.excel_importer)
+        types_changed = any(loc_type.needs_save for loc_type in type_stubs)
+        moved_to_root = any(loc.moved_to_root for loc in location_stubs)
+        delay_updates = not (types_changed or moved_to_root)
```

Comment:
it makes no improvents for those situations but i think they are quite rare. one if them is changes to the types hierarchy, which is easy enough to do from the ui since few people have more than a 10 if any, and if we warn people to make those changes first they easily can. the second exemption is a location moving from its current type to the root type, and it seems unlikely people will do that since locations should rarely change types, but again if that is necessary, they can move that type, then proceed with the rest of the upload.